src/com/phonebookservice/util/MyArrayList.java

```diff
@@ -392,4 +477,47 @@ private boolean checkIfEqual(final Object element, final int index) {
                 || this.data[index] != null && data[index].equals(element);
 
     }
+
+    private void checkCollectionNotNull(final Collection<?> collection) {
+        if (collection == null) {
+            throw new NullPointerException("collection is null");
+        }
+    }
+
+    private void increaseSizeInternal(final int minSize) {
+        int newIncreaseOfSize = minSize / (this.data.length - 1);
+        this.increaseSize(newIncreaseOfSize + 1);
+    }
+
+    private boolean removeCollection(final Collection<?> collection,
+            final boolean remove) {
+        this.checkCollectionNotNull(collection);
+        int changed = 0;
+        int index = 0;
+
+        for (int i = 0; i < currentIndex; i++) {
+            boolean contains = false;
+            boolean retain = true;
+            final Iterator<?> iterator = collection.iterator();
+
+            while (iterator.hasNext()) {
+                if ((iterator.next() == this.data[i])) {
+                    contains = true;
+                    retain = false;
```

Comment:
I recommend move this in private method call checkCollectionContains(T element, Collection )

U can use for each now as u understood iterator

Reply:
cant be added in a private method (not good) because I have 2 booleans to return to check if collection to remove or to retain